skrub/_table_vectorizer.py

```diff
@@ -458,6 +471,105 @@ def _clone_transformers(self):
 
         # TODO: check that the provided transformers are valid
 
+    def _split_univariate_transformers(self, split_fitted: bool = False):
+        """
+        Split univariate transformers into multiple transformers, one for each
+        column. This is useful to use the inherited `ColumnTransformer` class
+        parallelism.
+
+        Parameters
+        ----------
+        split_fitted : bool, default=False
+            Whether to split the self.transformers_ attribute (True) or the
+            self.transformers attribute (False). The former is used when
+            calling `transform` and the latter when calling `fit_transform`.
+        """
+        if split_fitted:
+            check_is_fitted(self, attributes=["transformers_"])
+            self._transformers_fitted_original = self.transformers_
+            new_transformers_ = []
+            new_transformer_to_input_indices = {}
+            for name, trans, cols in self.transformers_:
+                if (
+                    trans.__class__.__name__ in UNIVARIATE_TRANSFORMERS
+                    and len(cols) > 1
+                ):
+                    splitted_transformers_ = trans._split()
+                    assert len(splitted_transformers_) == len(cols)
+                    for i, col in enumerate(cols):
+                        new_transformers_.append(
+                            (
+                                name + "_split_" + str(i),
+                                splitted_transformers_[i],
+                                [col],
+                            )
+                        )
+                        new_transformer_to_input_indices[name + "_split_" + str(i)] = [
+                            self._transformer_to_input_indices[name][i]
+                        ]
+                else:
+                    new_transformers_.append((name, trans, cols))
+                    new_transformer_to_input_indices[
+                        name
+                    ] = self._transformer_to_input_indices[name]
+            self.transformers_ = new_transformers_
+            self._transformer_to_input_indices = new_transformer_to_input_indices
+        else:
+            self._transformers_original = self.transformers
```

Comment:
That's a good point! Do you think we should worry about this situation: the user fits the `TableVectorizer` a first time (successfully), then changes e.g `tb.low_card_cat_transformer`, then call `fit_transform` a second time. In this case, `_transformer_original` would not be updated with the new transformer. I can changing an attribute directly is not good practice, but the error would be really confusing to debug.

Reply:
Yes, good catch! I think we should worry about it because, as Gaël said, this can happen. 

One way to deal with it could be to define a `__setattr__` method where in case of any of our transformer attributes (e.g. `"low_card_cat_transformer"`) being passed, we could do:
```python
if hasattr(self, "_transformers_original"):
    delattr(self, "_transformers_original")
```
WDYT? I don't like playing with `__setattr__` or `__getattr__` much, so there's probably a better solution

Reply:
@Vincent-Maladiere I thought a bit more about this, and I think the original code should have worked: in `fit_transform`, we're always reconstructing `self.transformers` from the attributes `low_card_cat_transformer` etc., before splitting, which means that even if the `ColumnTransformer` fails before we can merge, the following `fit_transform` shouldn't be affected. Still, I like the solution we ended up implementing (without the additional `_transformers_original` attribute). In the meantime, I added tests to check for the two failure modes we were discussing, they may be useful if the implementation changes in the future